src/components/vacation-requests-table/vacation-requests-table.tsx

```diff
@@ -52,7 +55,7 @@ const VacationRequestsTable = ({
     adminMode ? allVacationRequestStatusesAtom : vacationRequestStatusesAtom
   );
   const containerRef = useRef(null);
-  const [rows, setRows] = useState<DataGridRow[]>([]);
+  const [rows, setRows] = useState<VacationsDataGridRow[]>([]);
```

Comment:
After taking another look this seems like that the components responsibilities are a bit off. You should instead pass the rows to this component as props and only keep the things like selected rows in this component state.

Reply:
Passing the rows as props now